sklearn/utils/_set_output.py

```diff
@@ -151,7 +154,10 @@ def is_supported_container(self, X):
         return isinstance(X, pl.DataFrame)
 
     def rename_columns(self, X, columns):
-        return X.rename(dict(zip(X.columns, columns)))
+        # we cannot use `rename` since it takes a dictionary and at this stage we have
+        # potentially duplicate column names in `X`
```

Comment:
does polars also allow duplicate names?

Reply:
Nop this is the reason that I will open a new PR to rename first name before renaming.
If we have duplicated columns then we need to raise a nice error message.